### PR TITLE
[staging-next] qt6.qtdeclarative: slightly less bad hack to make it build on Darwin again

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

Not sure why it broke, could be the recent cmake changes?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/qt-6/modules/qtdeclarative.nix [hunk 1]
```diff
@@ -4,24 +4,27 @@
...
     # prevent headaches from stale qmlcache data
     ../patches/0001-qtdeclarative-disable-qml-disk-cache.patch
     # add version specific QML import path
     ../patches/0002-qtdeclarative-also-use-versioned-qml-paths.patch
   ];
+
   cmakeFlags = [
     "-DQt6ShaderToolsTools_DIR=${pkgsBuildBuild.qt6.qtshadertools}/lib/cmake/Qt6ShaderTools"
+    # for some reason doesn't get found automatically on Darwin
+    "-DPython_EXECUTABLE=${lib.getExe pkgsBuildBuild.python3}"
```
Comment (SomeoneSerge) [inline pkgs/development/libraries/qt-6/modules/qtdeclarative.nix:27]: nit: `lib.cmakeFeature`

**Comment by paparodeo** [comment]
FWIW i was unable to reproduce on x64 darwin (monterey) building with and without the sandbox enabled.

**Comment by NickCao** [comment]
> FWIW i was unable to reproduce on x64 darwin (big sur) building with and without the sandbox enabled.

Might be on the wrong branch? This is targeting staging-next. https://hydra.nixos.org/eval/1808219?filter=qt6.qtdeclarative&compare=1808209&full=

File: pkgs/development/libraries/qt-6/modules/qtdeclarative.nix [hunk 2]
```diff
@@ -4,24 +4,27 @@
...
 , openssl
 , stdenv
-, python3
 , lib
 , pkgsBuildBuild
 }:
 
 qtModule {
   pname = "qtdeclarative";
-  strictDeps = !stdenv.isDarwin; # fails to detect python3 otherwise
```
Comment (NickCao) [inline pkgs/development/libraries/qt-6/modules/qtdeclarative.nix:14]: Any idea why?

Comment (K900) [reply pkgs/development/libraries/qt-6/modules/qtdeclarative.nix:14]: No idea, someone with an actual Darwin machine may want to investigate.

Comment (NickCao) [reply pkgs/development/libraries/qt-6/modules/qtdeclarative.nix:14]: https://github.com/nixOS/nixpkgs/commit/8c9c8ade2f88a85ccdd4858cc802d7b7d6c48fe0 looks suspicious.

**Comment by paparodeo** [comment]
> > > FWIW i was unable to reproduce on x64 darwin (big sur) building with and without the sandbox enabled.
> > 
> > 
> > Might be on the wrong branch? This is targeting staging-next. https://hydra.nixos.org/eval/1808219?filter=qt6.qtdeclarative&compare=1808209&full=
> 
> i did check i was on staging-next and that the cmake changes were in the tree i was using. will double check once i have access to the machine later.

```
$ git status
On branch staging-next
Your branch is up to date with 'origin/staging-next'.
$ git log -n 1 
commit 45080a955932de96b78c631fe9b3518a4e7bf8a8 (HEAD -> staging-next, origin/staging-next)
Merge: 12873fb718e4 2d2404abaf13
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Sat Aug 10 18:01:08 2024 +0000

    Merge master into staging-next
$ ls pkgs/by-name/cm/cmake/000-nixpkgs-cmake-prefix-path.diff
pkgs/by-name/cm/cmake/000-nixpkgs-cmake-prefix-path.diff
$ nix-build -A qt6.qtdeclarative
...
/nix/store/kasrw47ln7w6gs3a7v9plca4bvq0q2cd-qtdeclarative-6.7.2
$ uname -a
Darwin odel.lan 20.6.0 Darwin Kernel Version 20.6.0: Thu Jul  6 22:12:47 PDT 2023; root:xnu-7195.141.49.702.12~1/RELEASE_X86_64 x86_64 i386 Darwin
```

**Pull request merged**